spec/presenters/hyrax/iiif_manifest_presenter_spec.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+# rubocop:disable BracesAroundHashParameters maybe a rubocop bug re hash params?
+RSpec.describe Hyrax::IiifManifestPresenter do
+  subject(:presenter) { described_class.new(work) }
+  let(:work) { build(:monograph) }
+
+  describe 'manifest generation' do
+    let(:builder_service) { Hyrax::ManifestBuilderService.new }
+
+    it 'generates a IIIF presentation 2.0 manifest' do
+      expect(builder_service.manifest_for(presenter: presenter))
+        .to include('@context' => 'http://iiif.io/api/presentation/2/context.json')
+    end
+
+    context 'with file set and work members' do
+      let(:work) { create(:work_with_image_files) }
+
+      it 'generates a manifest with nested content' do
+        expect(builder_service.manifest_for(presenter: presenter)['sequences'].first['canvases'].count)
+          .to eq 2
```

Comment:
How do we get to this magic #? Is there a property of presenter or work that makes this 2?

Reply:
i think what's really needed in this case, and in https://github.com/samvera/hyrax/pull/4356#discussion_r436010991, is a set of matchers for `iiif_manifest`

```ruby
expect(my_manifest).to have_sequence(file_set_1, file_set_2, file_set_3, ...)
```

i'm reticent to spend time belaboring these specs, short of that; it felt to me like more precision here was going to be more fragile (`['sequences'].first['canvases'].count` is already a bit miserable).

one refinement could be `   .to eq work.file_set_ids.count`, but even that doesn't feel particularly helpful; it should really be `work.file_sets.select(:&image?).count` or something that starts to look a whole bunch like the implementation...

Reply:
I think this is a case for a code comment explaining the number two is the above logic. I'd certainly not want to run that query to determine that value.

Reply:
can add comments for both of these.

i'm serious about IIIF Manifest matchers, but thinking it's out-of-scope for now.